### PR TITLE
Adding common components for RIX

### PR DESCRIPTION
**Pull request by jyotiphy**
* Added following common components:
           -AL1K2
           -IM2K2
           -IM3K2
           -IM4K2
           -TM1K2
           -LI2K2-K2A
           -PF1K2
           -IM5K2
           -TM2K2
* LI2K2-K2A is laser out coupler, this program needs to be cross-checked

### CODE REVIEW

File: plc-kfe-rix-motion/_Config/IO/PLC Rail (EtherCAT).xti [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<TcSmItem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.beckhoff.com/schemas/2012/07/TcSmItem" TcSmVersion="1.0" TcVersion="3.1.4022.30" ClassName="CDevEtherCATDef" SubType="111">
+<?xml version="1.0"?>
+<TcSmItem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.beckhoff.com/schemas/2012/07/TcSmProject" TcSmVersion="1.0" TcVersion="3.1.4024.4" ClassName="CDevEtherCATDef" SubType="111">
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/_Config/IO/PLC Rail (EtherCAT).xti:2]: I notice the tc version changed to 4024.4, this is ok but make sure you're building with the same version that is running on the PLC

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_1_PlcTask.TcPOU [hunk 2]
```diff
@@ -12,10 +12,31 @@ PRG_IM1K1_PPM();
...
+PRG_LI2K2_K2A();
+PRG_PF1K2_WFS();
+PRG_IM5K2_PPM();
+PRG_TM2K2_ATM();
+
 
 PRG_3_PMPS_POST();
 PRG_4_LOG();]]></ST>
     </Implementation>
+    <LineIds Name="PRG_1_PlcTask">
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_1_PlcTask.TcPOU:29]: It looks like you're editing with the default line ids setting: https://confluence.slac.stanford.edu/display/PCDS/TwinCAT+3+Git+Setup+and+Best+Practices#TwinCAT3GitSetupandBestPractices-TwinCATLineIDSettings:LineIDs

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_AL1K2_L2SI.TcPOU [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        pv: AL1K2:L2SI
+        io: io
+    '}
+    {attribute 'TcLinkTo' := '.fbLaser.iLaserINT := TIIB[AL1K2-EL4004-E4]^AO Outputs Channel 1^Analog output;
+                              .fbLaser.iShutdownINT := TIIB[AL1K2-EL4004-E4]^AO Outputs Channel 2^Analog output'}
+    fbAL1K2: FB_REF;
+END_VAR]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbAL1K2.nTransitionAssertionID := 16#6140;
+fbAL1K2.nUnknownAssertionID := 16#6149;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_AL1K2_L2SI.TcPOU:16]: To match PMPS conventions all the IDs in this file should be in the range `16#612_`, the IDs chosen here are in use by AL1K4 and may cause issues.

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM2K2_PPM.TcPOU [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                              .fbYagThermoCouple.bError := TIIB[IM2K2-EL3314-E4]^TC Inputs Channel 2^Status^Error;
+                              .fbYagThermoCouple.bUnderrange := TIIB[IM2K2-EL3314-E4]^TC Inputs Channel 2^Status^Underrange;
+                              .fbYagThermoCouple.bOverrange := TIIB[IM2K2-EL3314-E4]^TC Inputs Channel 2^Status^Overrange;
+                              .fbYagThermoCouple.iRaw := TIIB[IM2K2-EL3314-E4]^TC Inputs Channel 2^Value'}                         
+    fbIM2K2: FB_PPM;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbIM2K2.nTransitionAssertionID := 16#2120;
+fbIM2K2.nUnknownAssertionID := 16#2129;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM2K2_PPM.TcPOU:26]: The PMPS id range for IM2K2 should be `16#222_`

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM3K2_PPM.TcPOU [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                              .fbYagThermoCouple.bError := TIIB[IM3K2-EL3314-E4]^TC Inputs Channel 2^Status^Error;
+                              .fbYagThermoCouple.bUnderrange := TIIB[IM3K2-EL3314-E4]^TC Inputs Channel 2^Status^Underrange;
+                              .fbYagThermoCouple.bOverrange := TIIB[IM3K2-EL3314-E4]^TC Inputs Channel 2^Status^Overrange;
+                              .fbYagThermoCouple.iRaw := TIIB[IM3K2-EL3314-E4]^TC Inputs Channel 2^Value'}                          
+    fbIM3K2: FB_PPM;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbIM3K2.nTransitionAssertionID := 16#2120;
+fbIM3K2.nUnknownAssertionID := 16#2129;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM3K2_PPM.TcPOU:26]: The PMPS id range should be `16#232_`, the current IDs here are in use by IM1K2

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM4K2_PPM.TcPOU [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                              .fbYagThermoCouple.bError := TIIB[IM4K2-EL3314-E4]^TC Inputs Channel 2^Status^Error;
+                              .fbYagThermoCouple.bUnderrange := TIIB[IM4K2-EL3314-E4]^TC Inputs Channel 2^Status^Underrange;
+                              .fbYagThermoCouple.bOverrange := TIIB[IM4K2-EL3314-E4]^TC Inputs Channel 2^Status^Overrange;
+                              .fbYagThermoCouple.iRaw := TIIB[IM4K2-EL3314-E4]^TC Inputs Channel 2^Value'}                         
+    fbIM4K2: FB_PPM;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbIM4K2.nTransitionAssertionID := 16#2120;
+fbIM4K2.nUnknownAssertionID := 16#2129;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM4K2_PPM.TcPOU:26]: The PMPS ID range should be `16#242_`, the current one here is in use by IM1K2

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM5K2_PPM.TcPOU [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                              .fbYagThermoCouple.bError := TIIB[IM5K2-EL3314-E4]^TC Inputs Channel 2^Status^Error;
+                              .fbYagThermoCouple.bUnderrange := TIIB[IM5K2-EL3314-E4]^TC Inputs Channel 2^Status^Underrange;
+                              .fbYagThermoCouple.bOverrange := TIIB[IM5K2-EL3314-E4]^TC Inputs Channel 2^Status^Overrange;
+                              .fbYagThermoCouple.iRaw := TIIB[IM5K2-EL3314-E4]^TC Inputs Channel 2^Value'}                           
+    fbIM5K2: FB_PPM;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbIM5K2.nTransitionAssertionID := 16#2120;
+fbIM5K2.nUnknownAssertionID := 16#2129;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_IM5K2_PPM.TcPOU:26]: PMPS ID range should be `16#252_`, this one is also in use by IM1K2

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_LI2K2_K2A.TcPOU [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+   1       1       1       1      1       1       1       1
+*)
+stSiBP.neVRange := 2#0000_0111_1111_1111;
+// Drop transmission to 20%
+stSiBP.nTran := 20;
+
+// Drop transmission to 10%
+stFoilBP.nTran := 10;
+
+fbLI2K2.nTransitionAssertionID := 16#4140;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_LI2K2_K2A.TcPOU:31]: PMPS IDs for LI2K2 should be `16#422_`, the ones here are from LI1K4

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_PF1K2_WFS.TcPOU [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+   2.3E3,  2.0E3,  1.82E3, 1.75E3, 1.65E3, 1.55E3, 1.45E3, 1.25E3,
+   1.15E3, 850,    540,    450,    400,    350,    270,    250
+   0       0       0       0      0       1       1       1
+   1       1       1       1      1       1       1       1
+*)
+stSiBP.neVRange := 2#0000_0111_1111_1111;
+// Drop transmission to 20%
+stSiBP.nTran := 20;
+
+fbPF1K2.nTransitionAssertionID := 16#3140;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_PF1K2_WFS.TcPOU:35]: PMPS IDs should be `16#312_`, the ID here is used by PF1K4

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_TM2K2_ATM.TcPOU [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    {attribute 'TcLinkTo' := '.fbThermoCouple1.bError := TIIB[TM2K2-EL3314-E5]^TC Inputs Channel 1^Status^Error;
+                              .fbThermoCouple1.bUnderrange := TIIB[TM2K2-EL3314-E5]^TC Inputs Channel 1^Status^Underrange;
+                              .fbThermoCouple1.bOverrange := TIIB[TM2K2-EL3314-E5]^TC Inputs Channel 1^Status^Overrange;
+                              .fbThermoCouple1.iRaw := TIIB[TM2K2-EL3314-E5]^TC Inputs Channel 1^Value'}
+    fbTM2K2: FB_ATM;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[fbTM2K2.nTransitionAssertionID := 16#5120;
+fbTM2K2.nUnknownAssertionID := 16#5129;
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/PRG_TM2K2_ATM.TcPOU:19]: IDs should be `16#522_`, this one is from TM1K2

**Comment by ZLLentz** [comment]
I pointed out everything that I saw, I tried to be thorough

**Comment by jyotiphy** [comment]
I have added all the new modifications. Can you please check and merge it into master ? Also, need to build IOC and deploy it to PLC.

**Review by ZLLentz — APPROVED** [review]
This looks good to me

**Pull request merged**